src/coreclr/src/pal/src/loader/module.cpp

```diff
@@ -592,6 +604,11 @@ PALAPI
 PAL_LoadLibraryDirect(
     IN LPCWSTR lpLibFileName)
 {
+    if (ShouldRedirectToCurrentLibrary(lpLibFileName))
```

Comment:
This should not need to check `g_running_in_exe` at all. We should be able to return `dlopen(NULL...)` for `lpLibFileName == NULL` uncoditionally.

Reply:
There is code following this treating NULL and "" as error conditions. I thought there is some value to fail on unexpected NULL.

Reply:
So should the  `g_running_in_exe`   be removed in the `ShouldRedirectToCurrentLibrary` helper?

If the check is unnecessary, I guess it is unnecessary in either case.

Reply:
Actually - I think the `g_running_in_exe` check could be an assert.